instrumentation/spring-webflux-5.0/src/main/java8/io/opentelemetry/auto/instrumentation/springwebflux/server/DispatcherHandlerAdvice.java

```diff
@@ -32,34 +34,38 @@
 public class DispatcherHandlerAdvice {
 
   @Advice.OnMethodEnter(suppress = Throwable.class)
-  public static SpanWithScope methodEnter(@Advice.Argument(0) final ServerWebExchange exchange) {
+  public static void methodEnter(
+      @Advice.Argument(0) final ServerWebExchange exchange,
+      @Advice.Local("otelScope") Scope otelScope,
+      @Advice.Local("otelContext") Context otelContext) {
     // Unfortunately Netty EventLoop is not instrumented well enough to attribute all work to the
-    // right things so we have to store span in request itself. We also store parent (netty's) span
-    // so we could update resource name.
-    final Span parentSpan = TRACER.getCurrentSpan();
-    if (parentSpan.getContext().isValid()) {
-      exchange.getAttributes().put(AdviceUtils.PARENT_SPAN_ATTRIBUTE, parentSpan);
-    }
+    // right things so we have to store the context in request itself.
+    // We also store parent (netty's) context so we could update resource name.
+    Context currentContext = Context.current();
+    exchange.getAttributes().put(AdviceUtils.PARENT_CONTEXT_ATTRIBUTE, currentContext);
 
     final Span span = TRACER.spanBuilder("DispatcherHandler.handle").startSpan();
```

Comment:
i think we can save a `Context.current()` lookup here since we have it already now:

`.setParent(TracingContextUtil.getSpan(parentContext))`